hack/common/Makefile

```
@@ -75,10 +75,17 @@ delete-k3d-cluster: ## Delete k3d kyma cluster.
 
 HIBERNATION_HOUR=$(shell echo $$(( ( $(shell date +%H | sed s/^0//g) + 5 ) % 24 )))
 GIT_COMMIT_SHA=$(shell git rev-parse --short=8 HEAD)
+ifeq (,$(GARDENER_K8S_VERSION))
+ifneq (,$(GARDENER_SA_PATH))
+GARDENER_K8S_VERSION=$(shell kubectl --kubeconfig=${GARDENER_SA_PATH} get cloudprofiles.core.gardener.cloud ${GARDENER_INFRASTRUCTURE} -o=jsonpath='{.spec.kubernetes.versions[0].version}')
+else
+GARDENER_K8S_VERSION=1.27.4
+endif
+endif
```

There is no need to check the `GARDENER_K8S_VERSION` value in the first if. Your can get rid of it to improve readability by adding `?` before all equal symbols. The idea in general is really smart :) I like it

```suggestion
ifneq (,$(GARDENER_SA_PATH))
GARDENER_K8S_VERSION?=$(shell kubectl --kubeconfig=${GARDENER_SA_PATH} get cloudprofiles.core.gardener.cloud ${GARDENER_INFRASTRUCTURE} -o=jsonpath='{.spec.kubernetes.versions[0].version}')
else
GARDENER_K8S_VERSION?=1.27.4
endif
```